components/dashboard/src/projects/Prebuilds.tsx

```diff
@@ -257,6 +257,15 @@ export function prebuildStatusIcon(prebuild?: PrebuildWithStatus) {
     }
 }
 
+function formatDuration(milliseconds: number) {
```

Comment:
In the past we used [moment](https://github.com/gitpod-io/gitpod/blob/7abe78c91792c18b0c22103c890b25ea3a4cdf05/components/dashboard/package.json#L10) for this kind of work. Why not re-use that?

Reply:
Hmm, `moment` doesn't seem that great for _formatting time differences_ specifically. (Also, their [docs](https://momentjs.com/docs/) suggest maybe not using it anymore in general.)

The primary advantage of this custom code is doing `mm:ss` when duration is less than 1 hour (e.g. `03:24`), and `h:mm:ss` when it's longer (e.g. `1:05:12`).

I guess we can achieve something similar with `moment` (using two different formats depending on >= or < 1 hour):

```ts
function formatDuration(milliseconds) {
    if (milliseconds < 1000 * 60 * 60) {
        return moment(milliseconds).format('mm:ss');
    }
    return moment(milliseconds).format('hh:mm:ss');
}
```

However (contrived edge case alert 🙈) the above code doesn't properly handle durations longer than 24 hours.

For example, for 25 hours (`milliseconds = 1000 * 60 * 60 * 25`):
- the `moment` code will return `01:00:00` (wrong!)
- the custom code will return `25:00:00` (still correct)

I guess we could handle this with yet another condition, but at this point the complexity with `moment` becomes similar to without `moment` 😋 (or, we just ignore the edge case, which is 99.99% okay but feels bad somehow).

Reply:
Maybe like this?

```ts
function formatDuration(milliseconds) {
    const hours = Math.floor(milliseconds / (1000 * 60 * 60));
    return (hours > 0 ? `${hours}:` : '') + moment(milliseconds).format('mm:ss');
}
```

This covers all the edge cases, is short, and uses `moment`. 👍